codegenerator/cli/templates/static/codegen/src/Config.res

```diff
@@ -97,21 +97,24 @@ let make = (
   ~entities=[],
   ~ecosystem=Evm,
 ) => {
+  let chainMap =
+    chains
+    ->Js.Array2.map(n => {
+      (n.chain, n)
+    })
+    ->ChainMap.fromArrayUnsafe
   {
     historyConfig: {
       rollbackFlag: shouldRollbackOnReorg ? RollbackOnReorg : NoRollback,
       historyFlag: shouldSaveFullHistory ? FullHistory : MinHistory,
     },
-    isUnorderedMultichainMode: Env.Configurable.isUnorderedMultichainMode->Option.getWithDefault(
-      Env.Configurable.unstable__temp_unordered_head_mode->Option.getWithDefault(
-        isUnorderedMultichainMode,
+    isUnorderedMultichainMode: chainMap->ChainMap.size === 1 ||
```

Comment:
Haven't finished reading the PR yet but wondering how this helps on a single chain.

Reply:
At least the batch creation logic is faster. I don't know about the entity history though. Should probably do the chainSize === 1 check in the actual batch creation instead.